lib/modules/manager/maven-wrapper/artifacts.ts

```diff
@@ -0,0 +1,200 @@
+      await getUpdatedArtifacts(status, artifactFileNames)
+    ).filter(is.truthy);
+
+    logger.debug(
+      { files: updateArtifactsResult.map((r) => r.file?.path) },
+      `Returning updated maven-wrapper files`
+    );
+    return updateArtifactsResult;
+  } catch (err) {
+    logger.debug({ err }, 'Error setting new Maven Wrapper release value');
+    return [
+      {
+        artifactError: {
+          lockFile: packageFileName,
+          stderr: err.message,
+        },
+      },
+    ];
+  }
+}
+
+async function getUpdatedArtifacts(
+  status: StatusResult,
+  artifactFileNames: string[]
+): Promise<(UpdateArtifactsResult | null)[]> {
+  const updatedResults: (UpdateArtifactsResult | null)[] = [];
+  for (const artifactFileName of artifactFileNames) {
+    updatedResults.push(await addIfUpdated(status, artifactFileName));
+  }
+  return updatedResults;
+}
+
+/**
+ * Find compatible java version for maven.
+ * see https://maven.apache.org/developers/compatibility-plan.html
+ * @param mavenWrapperVersion current maven version
+ * @returns A Java semver range
+ */
+export function getJavaConstraint(
+  mavenWrapperVersion: string | null | undefined
+): string | null {
+  const major = mavenWrapperVersion
+    ? mavenVersioning.getMajor(mavenWrapperVersion)
+    : null;
+
+  if (major && major >= 3) {
+    return '^17.0.0';
+  }
+
+  return '^8.0.0';
+}
+
+async function executeWrapperCommand(
+  cmd: string,
+  config: UpdateArtifactsConfig
+): Promise<void> {
+  logger.debug(`Updating maven wrapper: "${cmd}"`);
+  const execOptions: ExecOptions = {
+    docker: {
+      image: 'sidecar',
+    },
+    toolConstraints: [
+      {
+        toolName: 'java',
+        constraint:
+          config.constraints?.java ?? getJavaConstraint(config.currentValue),
+      },
+    ],
+  };
+
+  try {
+    await exec(cmd, execOptions);
+  } catch (err) {
+    logger.error({ err }, 'Error executing maven wrapper update command.');
+    throw err;
+  }
+}
+
+async function createWrapperCommand(): Promise<string | null> {
+  const projectDir = GlobalConfig.get('localDir');
+  const wrapperExecutableFileName = mavenWrapperFileName();
+  const wrapperFullyQualifiedPath = upath.resolve(
+    projectDir,
+    `./${wrapperExecutableFileName}`
+  );
+  return await prepareCommand(
+    wrapperExecutableFileName,
+    projectDir,
+    await statLocalFile(wrapperFullyQualifiedPath),
+    `wrapper:wrapper`
+  );
+}
+
+function mavenWrapperFileName(): string {
+  if (
+    os.platform() === 'win32' &&
+    GlobalConfig.get('binarySource') !== 'docker'
+  ) {
+    return 'mvnw.cmd';
+  }
+  return './mvnw';
+}
+
+async function prepareCommand(
+  fileName: string,
+  cwd: string | undefined,
+  pathFileStats: Stats | null,
+  args: string | null
+): Promise<string | null> {
+  // istanbul ignore if
+  if (pathFileStats?.isFile() === true) {
+    // if the file is not executable by others
+    if ((pathFileStats.mode & 0o1) === 0) {
```

Comment:
add a check for os, should not done on windows.

see latest Gradle wrapper changes